src/statement.d

```diff
@@ -1324,38 +1329,38 @@ extern (C++) class CompoundStatement : Statement
         return statements;
     }
 
-    override final ReturnStatement isReturnStatement()
+    override final inout(ReturnStatement) isReturnStatement() inout nothrow pure
```

Comment:
Can't just pick one of the statements and call it "is" a return statement. It isn't.  This works completely unlike all the other dynamic casting virtual functions.

That would be "has" a return statement.